abi/gen.go

```diff
@@ -0,0 +1,45 @@
+package abi
+
+import (
+	"bytes"
+	"go/format"
+	"text/template"
+)
+
+const ct = `
+// Generated by github.com/indexsupply/x/cmd/genabi. DO NOT EDIT.
+
+package {{.Package}}
+
+{{- range $path, $name := .Imports}}
+import {{$name}} {{printf "%q" $path}}
+{{- end}}
+
+{{range .Events}}
+type {{.Name}} struct {
+	{{- range .Inputs}}
+		{{.Name}} string
```

Comment:
Is `string` just a place holder for the actual go type? which we'll determine using a switch on `{{.Type}}`?

Reply:
Exactly. The next step here is to find a nice way of getting the correct go type for the input.